app/ControllerFunctions/ReadAccessFunctions.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+
+namespace App\ControllerFunctions;
+
+
+use App\Album;
+use App\ModelFunctions\SessionFunctions;
+use App\Photo;
+
+class ReadAccessFunctions
+{
+
+	/**
+	 * @var SessionFunctions
+	 */
+	private $sessionFunctions;
+
+
+
+	/**
+	 * @param SessionFunctions $sessionFunctions
+	 */
+	public function __construct(SessionFunctions $sessionFunctions)
+	{
+		$this->sessionFunctions = $sessionFunctions;
+	}
+
+
+
+	/**
+	 * Check if a (public) user has access to an album
+	 * if 0 : album does not exists
+	 * if 1 : access is granted
+	 * if 2 : album is private
+	 * if 3 : album is password protected and require user input
+	 *
+	 * @param $albumID
+	 * @return int
+	 */
+	public function albums($albumID)
+	{
+		// if we are logged in then we have access
+		// we do not use this function to check if there is proper acess in case of sharing... yet?
```

Comment:
s/acess/access/

Reply:
And yes, some proper access control would be good, but complicated and out of scope here.

Reply:
_proper access control_ is something I'm highly interested in working on but if somebody beats me to it, I won't complain of course :smiley:. This clean-up should obviously go in first anyway. I'm hoping to try it out tonight.